### PR TITLE
Add header

### PR DESCRIPTION
**Pull request by DrScientistPhD**
Lots of changes - check out the code and let me know what you think.

### CODE REVIEW

File: R/poke-winning.R [hunk 1]
```diff
@@ -1,47 +1,132 @@
...
+
+# Create a dataframe that calculates the relative weakness of each type 
+#   against each other type
+poke_df_weakness <- pokemon %>% 
+  select(type1, 
+         against_bug:against_water) %>% 
+  gather(defender,
+         effect,
+         against_bug:against_water) %>% 
+  mutate(defender = str_replace_all(defender, "against_", ""),
```
Comment (alex-antonison) [inline R/poke-winning.R]: You are labeling both as defender.  Would one of these be the attacker?

File: R/poke-winning.R [hunk 2]
```diff
@@ -1,47 +1,132 @@
...
+         defender = str_replace_all(defender, "Fight", "Fighting")) %>% 
+  group_by(defender,
+           type1) %>% 
+  summarize_all(funs(mean)) %>% 
+  group_by(defender) %>% 
+  mutate(relative_effectiveness = effect/max(effect) * 100)
```
Comment (alex-antonison) [inline R/poke-winning.R]: Should this be relative_weakness?

File: R/poke-winning.R [hunk 3]
```diff
@@ -1,47 +1,132 @@
...
-      as.data.frame() %>%
-      rownames_to_column() %>% 
-      rename("Type" = "rowname",
-             "Value" = "V1") 
-    
+  # Filter weakness data frame by selected type
+  filtered_poke_df_weakness <- reactive({
+    poke_df_type <- filter(poke_df_weakness, defender == input$Type) %>% 
+      select(attacker,
+             relative_weakness)
```
Comment (alex-antonison) [inline R/poke-winning.R]: Based on this, I believe the poke_df_weakness last mutate should be "relative_weakness"

File: R/poke-winning.R (hunk 1 above)
Comment (DrScientistPhD) [reply R/poke-winning.R]: No, I'm only relabeling the defender column to remove the "against_" text. I also needed to relabel the pokemon in the defender column that have the type "fight" to "fighting".

**Comment by DrScientistPhD** [comment]
The error in rendering the second plot appears linked to the choice input for the shiny app

**Comment by DrScientistPhD** [comment]
Alright, this shiny app is essentially done - let's have a final review of the code then we can merge to master.

**Pull request merged**